facebook-commerce-events-tracker.php

```diff
@@ -889,8 +889,13 @@ public function inject_purchase_event( $order_id ) {
 
 			foreach ( $order->get_items() as $item ) {
 
-				if ( $product = isset( $item['product_id'] ) ? wc_get_product( $item['product_id'] ) : null ) {
-
+				$product = null;
+				if ( isset( $item['variation_id'] ) ) {
+					$product = wc_get_product( $item['variation_id'] );
+				} else if ( isset( $item['product_id'] ) ) {
+					$product = wc_get_product( $item['product_id'] );
+				}
+				if ( $product ) {
```

Comment:
What do you think if we use `$item->get_product()` which will do the same check?

Code reference: https://github.com/woocommerce/woocommerce/blob/trunk/includes/class-wc-order-item-product.php#L326-L331

```suggestion
				$product = $item->get_product();
				
				if ( $product ) {
```

Reply:
Great suggestion - that's far better. I didn't know that existed.